File: ios/WalletBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WalletBridge, NSObject)

RCT_EXTERN_METHOD(createWallet: (RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(authenticateWithFaceID: (RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(signTransaction: (NSString *)transaction
                  resolve:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/appv3js/Bridge/WalletBridge.swift
import Foundation
import React
import ReactCommon

@objc(WalletBridge)
class WalletBridge: NSObject {
    
    @objc
    static func moduleName() -> String {
        return "WalletBridge"
    }
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    @objc
    func createWallet(_ resolve: @escaping (Any?) -> Void,
                     rejecter reject: @escaping (String, String, Error?) -> Void) {
        // Implementation will come later
    }
    
    @objc
    func authenticateWithFaceID(_ resolve: @escaping (Any?) -> Void,
                               rejecter reject: @escaping (String, String, Error?) -> Void) {
        // Implementation will come later
    }
    
    @objc
    func signTransaction(_ transaction: String,
                        resolve: @escaping (Any?) -> Void,
                        rejecter reject: @escaping (String, String, Error?) -> Void) {
        // Implementation will come later
    }
}
